src/components/AdminPanel/MedCentersForm/MedCentersForm.tsx

```diff
@@ -0,0 +1,154 @@
+  const [formState, setFormState] = useState<INewMedCenterData>(
+    isEdit
+      ? isEdit
+      : {
+          name: '',
+          address: '',
+          photo: '',
+          description: '',
+          services: [],
+        }
+  );
+
+  const dispatch = useDispatch();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const handleChange = (event: SelectChangeEvent<typeof formState.services>) => {
+    const {
+      target: { value },
+    } = event;
+    setFormState({
+      ...formState,
+      services: typeof value === 'string' ? value.split(',') : value,
+    });
+  };
+
+  const createNewCenter = (submitData: INewMedCenterData) => {
+    submitData.services = formState.services;
+    dispatch(submitFunction(submitData));
+  };
+
+  const editMedCenter = (submitData: INewMedCenterData) => {
+    submitData.services = formState.services;
+    submitData._id = isEdit?._id;
+    dispatch(submitFunction(submitData));
+  };
+
+  return (
+    <FormWrapper>
+      <Form onSubmit={handleSubmit(isEdit ? editMedCenter : createNewCenter)}>
+        <Input
+          primary
+          label="Name"
+          name="name"
+          register={register('name', Required)}
+          errors={errors}
+          value={formState.name}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+            setFormState({ ...formState, name: e.target.value })
+          }
+        />
+        <Input
+          primary
+          label="Address"
+          name="address"
+          register={register('address', Required)}
+          errors={errors}
+          value={formState.address}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+            setFormState({ ...formState, address: e.target.value })
+          }
+        />
+
+        <MedCenterPhotoWrapper>
+          <Input
+            primary
+            type="file"
+            name="photo"
+            register={register('photo', isEdit ? undefined : Photo)}
+            label="Select Photo"
+            errors={errors}
+            id={'photo'}
+          />
+          {isEdit && <img src={isEdit.photo} alt="medCenter Photo" />}
+        </MedCenterPhotoWrapper>
+
+        <TextArea
+          name="description"
+          register={register('description', Required)}
+          label="Description"
+          value={formState.description}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+            setFormState({ ...formState, description: e.target.value })
+          }
+        />
+
+        <FormSelect>
+          <FormSelectName>Services: </FormSelectName>
+          <FormControl sx={{ m: 1, width: 300 }}>
+            <Select
+              multiple
+              value={formState.services}
+              onChange={handleChange}
+              renderValue={(selected) => (
+                <Box sx={{ display: 'flex', flexWrap: 'wrap', gap: 0.5 }}>
```

Comment:
any way to avoid inline styles?

Reply:
https://github.com/KLIM-nk01/virtual-ambulance/pull/27/commits/d4619d031d9356f13c2fabbd27d63a4030c5cf6d#diff-82c3df5933f2e8057602e5ffc8dca1781e551a9ea622c252f31d3ac117e2fe1f